libgraph/src/analytics/k_shortest_paths/ksssp.cpp

```diff
@@ -0,0 +1,493 @@
+
+  katana::gPrint(" ", path->parent);
+}
+
+template <typename EdgeWeightType>
+static katana::Result<void>
+AddDefaultEdgeWeight(
+    katana::PropertyGraph* pg, const std::string& edge_weight_property_name, 
+    katana::TxnContext* txn_ctx) {
+  using EdgeData = std::tuple<EdgeWeightType>;
+
+  if (auto res = katana::analytics::ConstructEdgeProperties<EdgeData>(
+          pg, txn_ctx, {edge_weight_property_name});
+      !res) {
+    return res.error();
+  }
+
+  auto typed_graph = 
+      KATANA_CHECKED((katana::TypedPropertyGraph<std::tuple<>, EdgeData>::Make(
+          pg, {}, {edge_weight_property_name})));
+  katana::do_all(
+      katana::iterate(typed_graph.OutEdges()), 
+      [&](auto e) { typed_graph.template GetEdgeData<EdgeWeightType>(e) = 1; },
+      katana::steal(), katana::loopname("InitEdgeWeight"));
+  return katana::ResultSuccess();
+}
+
+template <typename GraphTy, typename Weight>
+katana::Result<void>
+KssspImpl(
+    GraphTy graph, uint32_t start_node, uint32_t report_node,
+    AlgoReachability algo_reachability, uint32_t num_paths, uint32_t step_shift,
+    kSsspPlan plan) {
+  using GNode = typename GraphTy::Node;
+
+  using kSSSP = katana::analytics::KSsspImplementationBase<
+      GraphTy, Weight, const Path, true>;
+  using kSSSPUpdateRequest = typename kSSSP::UpdateRequest;
+  using kSSSPUpdateRequestIndexer = typename kSSSP::UpdateRequestIndexer;
+  using kSSSPSrcEdgeTile = typename kSSSP::SrcEdgeTile;
+  using kSSSPSrcEdgeTilePushWrap = typename kSSSP::SrcEdgeTilePushWrap;
+  using kSSSPReqPushWrap = typename kSSSP::ReqPushWrap;
+  using kSSSPOutEdgeRangeFn = typename kSSSP::OutEdgeRangeFn;
+  using kSSSPTileRangeFn = typename kSSSP::TileRangeFn;
+
+  using OBIM =
+      katana::OrderedByIntegerMetric<kSSSPUpdateRequestIndexer, PSchunk>;
+  using OBIM_Barrier = typename katana::OrderedByIntegerMetric<
+      kSSSPUpdateRequestIndexer, PSchunk>::template with_barrier<true>::type;
+
+  using BFS = BfsSsspImplementationBase<GraphTy, unsigned int, false>;
+  using BFSUpdateRequest = typename BFS::UpdateRequest;
+  using BFSReqPushWrap = typename BFS::ReqPushWrap;
+  using BFSOutEdgeRangeFn = typename BFS::OutEdgeRangeFnUndirected;
+
+  auto it = graph.begin();
+  std::advance(it, start_node);
+  GNode source = *it;
+  it = graph.begin();
+  std::advance(it, report_node);
+  GNode report = *it;
+
+  size_t approxNodeData = graph.size() * 64;
+  katana::Prealloc(1, approxNodeData);
+  katana::ReportPageAllocGuard page_alloc;
+
+  katana::do_all(katana::iterate(graph), [&](const GNode& n) {
+    graph.template GetData<NodeMax<Weight>>(n) = 0;
+    graph.template GetData<NodeCount<Weight>>(n) = 0;
+  });
+
+  katana::StatTimer execTime("kSSSP");
+  execTime.start();
+
+  katana::InsertBag<std::pair<Weight, Path*>> paths;
+  katana::InsertBag<Path*> path_pointers;
+
+  bool reachable = true;
+
+  switch (algo_reachability) {
+  case async:
+    reachable = CheckReachabilityAsync<GraphTy, Weight, BFSUpdateRequest>(
```

Comment:
reachability should not be using weights

Reply:
There are values in reachability that have variable Weights, should I set them as constant?

Reply:
ok, let me check this

Reply:
no, it's not being used. I just checked this. Remove weight from reachability